Mathlib/AlgebraicTopology/SimplicialSet/HomotopyCat.lean

```diff
@@ -0,0 +1,345 @@
+        tgt_eq := by rw [← FunctorToTypes.naturality, f.tgt_eq] }
+    map_id := fun X ↦ OneTruncation₂.Hom.ext (by
+      dsimp
+      rw [← FunctorToTypes.naturality]) }
+
+@[ext]
+lemma OneTruncation₂.Hom_ext {S : SSet.Truncated 2} {x y : OneTruncation₂ S} {f g : x ⟶ y} :
+    f.edge = g.edge → f = g := OneTruncation₂.Hom.ext
+
+section
+variable {C : Type u} [Category.{v} C]
+
+/-- An equivalence between the type of objects underlying a category and the type of 0-simplices in
+the 2-truncated nerve. -/
+@[simps]
+def OneTruncation₂.nerveEquiv :
+    OneTruncation₂ ((SSet.truncation 2).obj (nerve C)) ≃ C where
+  toFun X := X.obj' 0
+  invFun X := .mk₀ X
+  left_inv _ := ComposableArrows.ext₀ rfl
+  right_inv _ := rfl
+
+/-- A hom equivalence over the function `OneTruncation₂.nerveEquiv`. -/
+def OneTruncation₂.nerveHomEquiv (X Y : OneTruncation₂ ((SSet.truncation 2).obj (nerve C))) :
+  (X ⟶ Y) ≃ (nerveEquiv X ⟶ nerveEquiv Y) where
+  toFun φ := eqToHom (congr_arg ComposableArrows.left φ.src_eq.symm) ≫ φ.edge.hom ≫
+      eqToHom (congr_arg ComposableArrows.left φ.tgt_eq)
+  invFun f :=
+    { edge := ComposableArrows.mk₁ f
+      src_eq := ComposableArrows.ext₀ rfl
+      tgt_eq := ComposableArrows.ext₀ rfl }
+  left_inv φ := by
+    ext
+    exact ComposableArrows.ext₁ (congr_arg ComposableArrows.left φ.src_eq).symm
+      (congr_arg ComposableArrows.left φ.tgt_eq).symm rfl
+  right_inv f := by dsimp; simp only [comp_id, id_comp]; rfl
+
+/-- The refl quiver underlying a nerve is isomorphic to the refl quiver underlying the category. -/
+def OneTruncation₂.ofNerve₂ (C : Type u) [Category.{u} C] :
+    ReflQuiv.of (OneTruncation₂ (nerveFunctor₂.obj (Cat.of C))) ≅ ReflQuiv.of C := by
+  apply ReflQuiv.isoOfEquiv.{u,u} OneTruncation₂.nerveEquiv OneTruncation₂.nerveHomEquiv ?_
+  intro X
+  unfold nerveEquiv nerveHomEquiv
+  simp only [Cat.of_α, op_obj, ComposableArrows.obj', Fin.zero_eta, Fin.isValue, Equiv.coe_fn_mk,
+    nerveEquiv_apply, Nat.reduceAdd, id_edge, SimplexCategory.len_mk, id_eq, eqToHom_refl, comp_id,
+    id_comp, ReflQuiver.id_eq_id]
+  unfold nerve truncation SimplicialObject.truncation SimplexCategory.Truncated.inclusion
+  simp only [fullSubcategoryInclusion.obj, SimplexCategory.len_mk, Nat.reduceAdd, Fin.isValue,
+    SimplexCategory.toCat_map, whiskeringLeft_obj_obj, Functor.comp_map, op_obj, op_map,
+    Quiver.Hom.unop_op, fullSubcategoryInclusion.map, ComposableArrows.whiskerLeft_map,
+    Fin.zero_eta, Monotone.functor_obj, Fin.mk_one, homOfLE_leOfHom]
+  show X.map (𝟙 _) = _
+  rw [X.map_id]
+  rfl
+
+/-- The refl quiver underlying a nerve is naturally isomorphic to the refl quiver underlying the
+category. -/
+@[simps! hom_app_obj hom_app_map inv_app_obj_obj inv_app_obj_map inv_app_map]
+def OneTruncation₂.ofNerve₂.natIso :
+    nerveFunctor₂.{u,u} ⋙ SSet.oneTruncation₂ ≅ ReflQuiv.forget := by
+  refine NatIso.ofComponents (fun C => OneTruncation₂.ofNerve₂ C) ?nat
```

Comment:
I think we may avoid `by refine...` here and just use `NatIso.ofComponents ... (by ....)`.